src/main/java/org/apache/ibatis/session/defaults/DefaultSqlSession.java

```diff
@@ -148,6 +148,7 @@ public <E> List<E> selectList(String statement, Object parameter, RowBounds rowB
   private <E> List<E> selectList(String statement, Object parameter, RowBounds rowBounds, ResultHandler handler) {
     try {
       MappedStatement ms = configuration.getMappedStatement(statement);
+      dirty |= ms.isDirtySelect();
```

Comment:
@harawata The `selectCursor` is not need to support dirty select?

Reply:
Totally missed it. Nice save, thanks!